src/traversals/eulerian.jl

```diff
@@ -0,0 +1,93 @@
+# Adapted from SimpleGraphs.jl [Copyright (c) 2014, Ed Scheinerman]:
+#    https://github.com/scheinerman/SimpleGraphs.jl/blob/master/src/simple_euler.jl
+# Reproduced under the MIT Expat License.
+
+"""
+    eulerian(g::AbstractSimpleGraph{T}[, u::T]) --> T[]
+
+Returns a [Eulerian trail or cycle](https://en.wikipedia.org/wiki/Eulerian_path) through an
+undirected graph `g`, starting at `u`, returning a vector listing the vertices of `g` in the
+order that they are traversed. If no such trail or cycle exists, throws an error.
+
+## Optional arguments
+- If `u` is omitted, a Eulerian trail or cycle is computed with `u = first(vertices(g))`.
+"""
+function eulerian(g::AbstractSimpleGraph{T}, u::T=first(vertices(g))) where {T}
+    is_directed(g) && error("`eulerian` is not yet implemented for directed graphs")
+    
+    _check_eulerian_input(g, u) # perform basic sanity checks
+
+    g′ = SimpleGraph{T}(nv(g)) # copy `g`
```

Comment:
Why do you need a copy?

Reply:
The algorithm below is relying on removing edges from the graph as it is doing the traversal; i.e., it destructively modifies the graph.